### DISCUSSION TITLE
Real-time problem

### DISCUSSION THREAD

**Discussion by tugbayk** [nikosT/Gisola#44] [General]
Dear  @nikosT 
I installed and ran the Gisola program. It is very successful in MT solutions of past earthquakes, but I could not run it in Real-time. 
First of all 
./gisola.py --datetime-range 2024-03-13T00:00:00 2024-03-14T09:00:00 -c configafd.yaml

2024-03-14 08:50:59,623 INFO Connecting to FDSNWS-event host: http://127.0.0.1:8080
2024-03-14 08:50:59,642 INFO Requesting event in timespan: 2024-03-13T00:00:00.000000Z - 2024-03-14T09:00:00.000000Z
2024-03-14 08:50:59,643 INFO Minimum Magnitude and type is set to: 3.8 and MLv respectively
2024-03-14 08:50:59,963 INFO Starting MT calculation for Event Event: 2024-03-14T03:07:54.280208Z | +41.601, +24.357 | 4.933841127 M | automatic

When the above command is executed, it immediately finds the event and switches to the MT solution. 

crontab definition  
     * * * * * * * source /home/sysop/.bashrc; cd /home/sysop/Gisola/src; python3 ./gisola.py --real-time -c configafd.yaml;

~/Gisola/src$ gisola.py --real-time -c configafd.yaml  

I tried typing the command manually at the moment of the earthquake, but fdsnws cannot find the event. 

My watcher settings are as follows 
 
Watcher:
  Magnitudetype: MLv # set specific hypocenter magnitude type or null for all
  Range: 540 # check interval in sec
  Playback: 0 # set sec for past-time run
  Historical: false # use this only if Playback is not zero and you want to reproduce the exact scenario
  Geobox: null

Is there something I'm missing?

Thanks a lot

Best wishes,
Tugbay

**Comment by nikosT** [comment]
Dear @tugbayk ,
From what you posted, I do not see anything problematic. From what I understand the issue exists at the triggering level and not at the computation part. So we need to see why Watcher cannot see the event. Various things could be responsible for it.
First, check the logs of Gisola, to see at what time it performs the FDSNWS-event check. Usually, a PC with a misconfigured UTC time could affect the outcome. Second, it might need more time than default playback to push the event to FDSNWS-event. So, check when an event happens, how much time is it needed to recognize it and put it in FDSNWS? Probably some auto-loc and auto-origin routines might take some time until they push it to the FDSNWS.  If this is the case, you can set a Playback value of X seconds to check in the past. If you put a large value you might miss the real-time operation that case, so you have to be cautious. So please, check the watcher's logs, to see at what time it checks for an event and also if that event is pushed to FDSNWS at that time.

**Reply by tugbayk** [reply]
Dear nikosT, 

thank you for your help, I will check what you said. 
Best wishes,
 
Tugbay

**Comment by tugbayk** [comment]
I think I found the reason why it doesn't work automatically in real time. 
I put into crontab ;
source /home/sysop/.bashrc; cd /home/sysop/Gisola/src; python3 ./gisola.py --real-time -c configafd.yaml;
when I run this command terminal, 
(gisola) sysop@sL0-ScAppTK04:~/Gisola/src$ source /home/sysop/.bashrc; cd /home/sysop/Gisola/src; python3 ./gisola.py --real-time -c configafd.yaml;
Backtrace (last call):
  File "/home/sysop/Gisola/src/./gisola.py", line 29, in <module>
    import event, inventory, flow, flow, isolate, plot, web, configuration
  File "/home/sysop/Gisola/src/event.py", line 27, in <module>
    from turfpy.measurement import boolean_point_in_polygon
ModuleNotFoundError: There is no module called 'turfpy'
(base) sysop@sL0-ScAppTK04:~/Gisola/src$ 
As you can see conda is coming out of the activated gisola.

I then extracted (source /home/sysop/.bashrc;) and it works like this.  

(gisola)sysop@sL0-ScAppTK04:~/Gisola/src$ cd /home/sysop/Gisola/src; python3 ./gisola.py --real-time -c configafd.yaml;
2024-03-25 14:19:48,819 INFO Connecting to FDSNWS-event host: http://127.0.0.1:8080
2024-03-25 14:19:48,841 INFO Requesting event in time slot: 2024-03-25T13:59:50.819714Z - 2024-03-25T14:15:40.819714Z
2024-03-25 14:19:48,841 INFO is set to Minimum Size and type: 3,8 and MLv respectively
2024-03-25 14:19:48,848 INFO Event not found...
2024-03-25 14:19:48,849 INFO 0 Event(s) in catalog:

I think I found the reason why it doesn't work automatically in real time. 
I looked in .bahrc and this is the only thing about gisola.  I don't know if there is something wrong here. 

################Gisola ##################
ulimit -s unlimited 
export PATH="/home/sysop/Gisola/src:$PATH"

I'm in crontab 
cd /home/sysop/Gisola/src; python3 ./gisola.py --real-time -c configafd.yaml; I updated it this way.  Will there be any problems?

Thanks a lot 

Best wishes 

Tugbay

**Comment by nikosT** [comment]
Hi @tugbayk ,
I didn't understand what you think is the problem. Indeed it would be best if you were sure that conda env is activated correctly during the cron job. However, according to the log you posted, the search iteration for new events is happening without problems, since no error occurred. The fact that no events are found is another story. I believe the error is in time. What I would suggest is to set a very large time window for new events, to be sure to catch up with any of them. 
So modify `Range: 540` to have a big value e.g. 3 days: 10800 sec. You can also set the offset to move the time window to the past to be sure that you will find events. I would consider this as the first step. Then, I would try to understand why the initial was not working and what values should be set for this. Perhaps, the system that throws the events (FDSNWS) takes a lot of time to distribute them, so then the values should be set accordingly.
